### ISSUE TITLE
[Bug]: unregistration hook in menus doesn't work

### ISSUE THREAD

**Issue by RiverVanRain** [Elgg/Elgg#14319]
### Elgg version

4.3.8

### Source of your Elgg installation

Elgg.org website

### Describe the bug

Install and activate [Static](https://github.com/ColdTrick/static/releases/download/v10.1.2/static_v10.1.2.zip) plugin.

In your custom plugin `elgg-plugin.php` add this hook:
```
'hooks' => [
        'register' => [
         'menu:owner_block' => [
		'\ColdTrick\StaticPages\Menus::ownerBlockMenuRegister' => ['unregister' => true],
	],
 ],
```
This doesn't work, the menu is still displayed.

### Steps to reproduce

1.
2.
3.
...

**Comment by RiverVanRain** [comment]
These issues are many - the `unregister` hook doesn't work correctly.

**Comment by jdalsem** [comment]
Why did you close this?

**Comment by RiverVanRain** [comment]
I don't have enough tests to check this issue.
Sometimes it works, sometimes it doesn't.

For example, yesterday I found that this event is missing on my Elgg instance:
`'create', 'group' => 'Elgg\Groups\Group::createAccessCollection',`

This was due to the fact that I unregistered this event in my custom plugin:
`'join', 'group' => 'Elgg\Groups\Group::joinGroup' => ['unregister' => true],`

To restore it I had to do the following with same code:
```
'create' => [
	'group' => [
		'Elgg\Groups\Group::createAccessCollection' => ['unregister' => true],
		'\MyPlugin\Groups\Events::createAccessCollection' => [],
	],
```

I don't know why it's happening.
